File: Python/queue/BlackBox/BlackBox.py
import random
class Node:
    """Represents a single node in the queue."""
    def __init__(self, value):
        self.value = value
        self.next = None

class Queue:
    """A First-In-First-Out (FIFO) queue implementation."""
    def __init__(self):
        self.front = None
        self.rear = None
        self.size = 0

    def is_empty(self):
        """Checks if the queue is empty."""
        return self.size == 0

    def enqueue(self, value):
        """Adds an element to the end of the queue."""
        node = Node(value)
        if self.rear is None:
            self.front = self.rear = node
        else:
            self.rear.next = node
            self.rear = node
        self.size += 1

    def dequeue(self):
        """Removes an element from the front of the queue."""
        if self.is_empty():
            raise IndexError("Cannot dequeue from an empty queue")
        value = self.front.value
        self.front = self.front.next
        if self.front is None:
            self.rear = None
        self.size -= 1
        return value

    def peek(self):
        """Returns the element at the front of the queue without removing it."""
        if self.is_empty():
            raise IndexError("Cannot peek an empty queue")
        return self.front.value

    def __len__(self):
        """Returns the number of elements in the queue."""
        return self.size

    def __repr__(self):
        """Returns a string representation of the queue."""
        values = []
        current = self.front
        while current:
            values.append(str(current.value))
            current = current.next
        return " -> ".join(values)

def main():
    q = Queue()
    # Enqueue 1000 random values
    for _ in range(1000):
        q.enqueue(random.randint(1, 10000))

    # Dequeue all items
    while not q.is_empty():
        q.dequeue()

    # Attempt to dequeue from an empty queue (will raise an exception)
    try:
        q.dequeue()
    except IndexError:
        pass  # Handling the exception gracefully

if __name__ == "__main__":
    main()

File: com/Java/queue/BlackBox/BlackBox.java
import java.util.NoSuchElementException;
public class BlackBox<T> {
    private Node<T> front; // front of the queue
    private Node<T> rear; // rear of the queue
    private int size; // number of elements in the queue

    // Node class for the linked list
    private static class Node<T> {
        T data;
        Node<T> next;

        public Node(T data) {
            this.data = data;
            this.next = null;
        }
    }

    // constructor
    public BlackBox() {
        front = null;
        rear = null;
        size = 0;
    }

    // enqueue operation (add element to the rear of the queue)
    public void enqueue(T element) {
        Node<T> newNode = new Node<>(element);
        if (rear == null) {
            front = newNode;
            rear = newNode;
        } else {
            rear.next = newNode;
            rear = newNode;
        }
        size++;
    }

    // dequeue operation (remove element from the front of the queue)
    public T dequeue() {
        if (front == null) {
            throw new NoSuchElementException("Queue is empty");
        }
        T element = front.data;
        front = front.next;
        if (front == null) {
            rear = null;
        }
        size--;
        return element;
    }

    // peek operation (return the front element without removing it)
    public T peek() {
        if (front == null) {
            throw new NoSuchElementException("Queue is empty");
        }
        return front.data;
    }

    // size operation (return the number of elements in the queue)
    public int size() {
        return size;
    }

    // isEmpty operation (check if the queue is empty)
    public boolean isEmpty() {
        return size == 0;
    }
    public static void main(String[] args) {
        // Create an instance of BlackBox for Integer type
        BlackBox<Integer> queue = new BlackBox<>();

        // Generate 1000 random values and enqueue them
        Random rand = new Random();
        int size = 1000;

        for (int i = 0; i < size; i++) {
            queue.enqueue(rand.nextInt(10000)); // Random values between 0 and 9999
        }

        // Dequeue all values to an array
        Integer[] array = new Integer[size];
        for (int i = 0; i < size; i++) {
            array[i] = queue.dequeue();
        }

        // Sort the array using merge sort
        mergeSort(array);

        // No print statements as requested
    }
}